pool/connection_pool.go

```diff
@@ -1109,8 +1124,16 @@ func (p *ConnectionPool) handlerDeactivated(conn *tarantool.Connection,
 	}
 }
 
-func (p *ConnectionPool) fillPools() bool {
+func (p *ConnectionPool) deactivateConnection(addr string,
+	conn *tarantool.Connection, role Role) {
+	p.deleteConnection(addr)
+	conn.Close()
+	p.handlerDeactivated(conn, role)
+}
+
+func (p *ConnectionPool) fillPools(ctx context.Context) (bool, bool) {
```

Comment:
The method looks too complicated. Please, try to split it into several methods.

Reply:
Updated.